imap_processing/glows/l1/glows_l1a_data.py

```diff
@@ -0,0 +1,411 @@
+import struct
+from dataclasses import dataclass
+from typing import ClassVar
+
+from imap_processing.glows import version
+from imap_processing.glows.l0.glows_l0_data import DirectEventL0, HistogramL0
+from imap_processing.glows.utils.constants import DirectEvent, GlowsConstants, TimeTuple
+
+
+class DataEverySecond:
+    """Mapping of name to byte for dataeverysecond structure in GLOWS L1A.
+
+    This is used to generate the housekeeping info for each direct event from the
+    compressed structure in the first 40 bytes of each direct event data field.
+
+    Attributes
+    ----------
+    mapping: dict
+        Mapping of value name to the number of bytes in the compressed data
+
+    """
+
+    mapping: ClassVar[dict] = {
+        "imap_sclk_last_pps": 4,
+        "glows_sclk_last_pps": 4,
+        "glows_ssclk_last_pps": 4,
+        "imap_sclk_next_pps": 4,
+        "catbed_heater_active": 1,
+        "spin_period_valid": 1,
+        "spin_phase_at_next_pps_valid": 1,
+        "spin_period_source": 1,
+        "spin_period": 2,
+        "spin_phase_at_next_pps": 2,
+        "number_of_completed_spins": 4,
+        "filter_temperature": 2,
+        "hv_voltage": 2,
+        "glows_time_on_pps_valid": 1,
+        "time_status_valid": 1,
+        "housekeeping_valid": 1,
+        "is_pps_autogenerated": 1,
+        "hv_test_in_progress": 1,
+        "pulse_test_in_progress": 1,
+        "memory_error_detected": 1,
+    }
+
+
+@dataclass
+class HistogramL1A:
+    """Data structure for GLOWS Histogram Level 1A data.
```

Comment:
btw, thank you for great documentation! Makes it easy to follow algorithm that we have not read through.